server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java

```diff
@@ -0,0 +1,104 @@
+package org.infinispan.rest.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JAVASCRIPT;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON_TYPE;
+import static org.infinispan.commons.dataconversion.MediaType.TEXT_PLAIN_TYPE;
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.POST;
+import static org.infinispan.rest.framework.Method.PUT;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.dataconversion.StandardConversions;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.scripting.ScriptingManager;
+import org.infinispan.tasks.Task;
+import org.infinispan.tasks.TaskContext;
+import org.infinispan.tasks.TaskManager;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * @since 10.1
+ */
+public class TasksResource implements ResourceHandler {
+   private final InvocationHelper invocationHelper;
+
+   public TasksResource(InvocationHelper invocationHelper) {
+      this.invocationHelper = invocationHelper;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            .invocation().methods(GET).path("/v2/tasks/").handleWith(this::listTasks)
+            .invocation().methods(PUT, POST).path("/v2/tasks/{taskName}").handleWith(this::createScriptTask)
+            .invocation().methods(GET).path("/v2/tasks/{taskName}").withAction("exec").handleWith(this::runTask)
+            .create();
+   }
+
+   private CompletionStage<RestResponse> listTasks(RestRequest request) {
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      TaskManager taskManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(TaskManager.class);
```

Comment:
is it possible the taxmanager to be null ?

Reply:
Yes if the tasks library is missing. But then you cannot invoke any of the /tasks endpoints either